### ISSUE TITLE
docker-machine does not support setting the docker engine bridge IP

### ISSUE THREAD

**Issue by kelseyhightower** [docker/machine#1455]
Currently docker-machine times out when setting the bip flag.

```
docker-machine create -d vmwarefusion --engine-opt bip=10.200.0.1/24 node0
Creating SSH key...
Creating VM...
Starting node0...
Waiting for VM to come online...
Error creating machine: Maximum number of retries (60) exceeded
You will want to check the provider to make sure the machine and associated resources were properly removed.
```

**Comment by nathanleclaire** [comment]
Dang, should work.  I'll try / run by our in house networking gurus and get back to you.

**Comment by nathanleclaire** [comment]
Hm, OK, so I know why this happens, but not 100% sure how to fix it.  Basically, the first time docker comes up it creates `docker0` with the default `bip`, but then we try to restart it with the new parameters and it doesn't see the network the right way.  You have to delete the default bridge and restart Docker to get it to work.

**Comment by odedpr** [comment]
Hi, 

I also got this issue, 

I ran this command 
<docker-machine create docker -d virtualbox --engine-opt bip=10.20.0.0/16>

**Comment by denverdino** [comment]
One walk around is to patch the machine/libmachine/provision/utils.go
Add the following lines after Line 169

```
	// Tempory fix
	if _, err = p.SSHCommand("ip link delete docker0"); err != nil {
		return err
	}
```
It will remove the docker0 bridge after set the docker engine options. And when docker daemon is started, it will create a new docker0 bridge based on the  "--bip" settings.

It works on Ubuntu and should work on RHEL/CentOS. But I cannot make it general for all the supported OS in the provisioners.

Thanks

**Comment by brianm** [comment]
#1662 is a dupe of this (with some additional detail)

**Comment by nathanleclaire** [comment]
interesting strategy @danverdino .  I wonder if it would cause any other issues.  I'm in favor if it seems safe.

@mavenugo @mrjana Do you have any input on this thread?  I'm wondering if @danverdino solution seems reasonable.

**Comment by byxorna** [comment]
Deleting (and having docker recreate) the `docker0` interface can be avoided if you ensure the `EXTRA_ARGS` are populated with all the engine options before starting up for the first time. I haven't dug in, but ensuring `/var/lib/boot2docker/profile` is present before ever launching docker makes the bridge interface come up with the right IP.

**Comment by brianm** [comment]
Was this fixed? I haven't seen an update that has better behavior, yet :-(

**Comment by chrisleck** [comment]
I'm also running into this.  Any ETA on a fix?

**Comment by adambiggs** [comment]
+1

**Comment by puleos** [comment]
I am running docker machine 0.6.0 and it appears to still be an issue.

**Comment by nathanleclaire** [comment]
@puleos Can you open another issue and post `--debug` output please?  Thanks!

**Comment by Starefossen** [comment]
@nathanleclaire I have the same problem as OP, unable to set docker0 IP when creating a new machine. I have created a new issue with debug output as per your request.

**Comment by Starefossen** [comment]
As a workaround I was able to add `bip` and `fixed-cidr` to the boot2docker configuration like this:

```
docker-machine ssh
vi /var/lib/boot2docker/profile 
```

and modified `EXTRA_ARGS` like this:

```
EXTRA_ARGS='
--label provider=vmwarefusion
--bip=192.168.1.5/24
--fixed-cidr=192.168.1.0/25
'
```

**Comment by drnybble** [comment]
As shown above:
  docker-machine create docker -d virtualbox --engine-opt bip=10.20.0.0/16
Still fails to create the VM with docker-machine 0.9.0
